### PR TITLE
osu-lazer: 2023.301.0 -> 2023.305.0

### PR DESCRIPTION
**Pull request by stepbrobd**
###### Description of changes

This PR is a duplicate of #219954, the original PR is inactive for two weeks and I have made changes to fix the update script and some styling changes based on the reviews in the original PR.

1. https://github.com/NixOS/nixpkgs/pull/219954#discussion_r1129983988

Extracted the osu-lazer-bin update script from update.sh to update-bin.sh 

2. https://github.com/NixOS/nixpkgs/pull/219954#discussion_r1129987407

Made the file url and hash to be in the same attrset.

3. https://github.com/NixOS/nixpkgs/pull/219954#issuecomment-1460850713

Fixed incorrect hash produced by `nix-prefetch-url` for darwin systems.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by thiagokokada — COMMENTED** [review]
Code wise looks good (did not test yet), but please squash your commits.

In this case I think the best way would be to have 3 commits:
```
osu-lazer: 2023.301.0 -> 2023.305.0
osu-lazer-bin: 2023.301.0 -> 2023.305.0
osu-lazer-bin: adding support for darwin systems
```

**Comment by StepBroBD** [comment]
> osu-lazer: 2023.301.0 -> 2023.305.0
osu-lazer-bin: 2023.301.0 -> 2023.305.0
osu-lazer-bin: adding support for darwin systems

Thanks for the review! Just squashed the commits.

**Review by thiagokokada — COMMENTED** [review]
There are 2 commits for "adding support for darwin systems".

Can you squash both in the same one? If you want to keep credits you can use something like Co-Authored-By.

**Comment by StepBroBD** [comment]
Sorry about that, should be fixed now.

**Review by thiagokokada — APPROVED** [review]

**Comment by delan** [comment]
Thanks! Sorry about #219954, didn’t have time to fix the hashing problem.

File: pkgs/games/osu-lazer/bin.nix [hunk 1]
```diff
@@ -1,37 +1,83 @@
...
+, fetchzip
+, appimageTools
+}:
 
-appimageTools.wrapType2 rec {
+let
   pname = "osu-lazer-bin";
-  version = "2023.301.0";
+  version = "2023.305.0";
+  name = "${pname}-${version}";
```
Comment (Lord-Valen) [inline pkgs/games/osu-lazer/bin.nix:11]: Why was this necessary?

Comment (stepbrobd) [reply pkgs/games/osu-lazer/bin.nix:11]: Nice catch! Nope, `name = "${pname}-${version}";` should be removed in the next update, or you can make a separate PR removing it.

Comment (Lord-Valen) [reply pkgs/games/osu-lazer/bin.nix:11]: It was bothering me because I use `allowUnfreePredicate`, and this causes it to change names (thus giving an unfree error) with each update. Nice to know that it was superfluous.
